Sources/SpeziQuestionnaire/TimedWalkView.swift

```diff
@@ -0,0 +1,99 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2022 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import FHIRQuestionnaires
+import ModelsR4
+import ResearchKit
+import ResearchKitOnFHIR
+import SwiftUI
+
+
+/// Renders a FHIR `Questionnaire`.
+public struct TimedWalkView: View {
+    @EnvironmentObject private var questionnaireDataSource: QuestionnaireDataSource
+    
+    @Binding private var isPresented: Bool
+    
+    private let identifier: String
+    private let distanceInMeters: Double
+    private let timeLimit: TimeInterval
+    private let turnAroundTimeLimit: TimeInterval
+    private let timedWalkResponse: ((ORKTimedWalkResult) async -> Void)?
+
+        
+    public var body: some View {
+        if let task = createTask() {
+            ORKOrderedTaskView(
+                tasks: task,
+                isPresented: $isPresented,
+                timedWalkResponse: { response in
+                    await timedWalkResponse?(response)
+                    await questionnaireDataSource.add(response)
+                },
+                tintColor: .accentColor
+            )
+                .ignoresSafeArea(.container, edges: .bottom)
+                .ignoresSafeArea(.keyboard, edges: .bottom)
+        } else {
+            Text("QUESTIONNAIRE_LOADING_ERROR_MESSAGE")
+        }
+    }
+    
+    
+    /// - Parameters:
+    ///   - questionnaire: The  `Questionnaire` that should be displayed.
+    ///   - isPresented: Indication from the questionnaire view if should be presented (not "Done" pressed or cancelled).
+    ///   - completionStepMessage: Optional completion message that can be appended at the end of the questionnaire.
+    ///   - questionnaireResponse: Optional response closure that can be used to manually obtain the `QuestionnaireResponse`.
+    public init(
+        identifier: String,
```

Comment:
I would suggest that we provide a reasonable default here that just generates a unique string for every time that the task is generated:
```suggestion
        identifier: String = UUID().uuidString,
```